extensions-core/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcExtensionsModule.java

```diff
@@ -81,7 +75,15 @@ public void configure(Binder binder)
     finally {
       Thread.currentThread().setContextClassLoader(currCtxCl);
     }
+  }
 
+  @Override
+  public void configure(Binder binder)
+  {
+    // this block of code is common among extensions that use Hadoop things but are not running in Hadoop, in order
+    // to properly initialize everything
+
+    final Configuration conf = new Configuration();
```

Comment:
wouldn't it still cause the same problem?

Reply:
The problem is being caused by the `FileSystem.get(conf)` as DFS Client is getting initialized with `fs.defaultFS` hostname, and during startup, the Hadoop cluster doesn't exist (hostname DNS not getting resolved) hence moved that part out of the configure function that is getting called during startup.